File: robocode-tankroyale-server/src/main/java/dev/robocode/tankroyale/server/model/Round.java
package dev.robocode.tankroyale.server.model;

import java.util.List;

import lombok.Builder;
import lombok.Singular;
import lombok.Value;

/**
 * State of a game round.
 * 
 * @author Flemming N. Larsen
 */
@Value
@Builder(toBuilder=true)
public final class Round {

	/** Round number */
	int roundNumber;

	/** List of turns */
	@Singular List<Turn> turns;

	/** Flag specifying if round has ended */
	boolean roundEnded;
	
	/**
	 * Returns the last turn of this round
	 */
	public Turn getLastTurn() {
		int numTurns = turns.size();
		if (numTurns > 0) {
			return turns.get(numTurns - 1);
		}
		return null;
	}
}

File: robocode-tankroyale-server-kotlin/src/main/kotlin/dev/robocode/tankroyale/server/model/Round.kt
package dev.robocode.tankroyale.server.model

/** Immutable state of a round in a battle. */
data class Round(
    /** Round number */
    override val roundNumber: Int,

    /** List of turns */
    override val turns: List<ITurn>,

    /** Flag specifying if round has ended yet */
    override val roundEnded: Boolean,

) : IRound